CLI.md

```diff
@@ -0,0 +1,105 @@
+# Command Line Interface
+
+This covers more advanced CLI use cases and workflows. To [get started](README.md#getting-started) with the CLI, reference 
+the [readme](README.md). Basic use of the CLI such as [creating a rule](CONTRIBUTING.md#creating-a-rule-with-the-cli) or 
+[testing](CONTRIBUTING.md#testing-a-rule-with-the-cli) are referenced in the [contribution guide](CONTRIBUTING.md).
+
+
+## Using a config file or environment variables
+
+CLI commands which are tied to Kibana and Elasticsearch are capable of parsing auth-related keyword args from a config 
+file or environment variables. Any command option which calls `callback=set_param_values` will search for set kwargs.
```

Comment:
I think we should assume the reader isn't familiar with the python here and spell things out as they apply to using the CLI

Reply:
standardized the args a bit better and added a list of them under `config` section of CLI doc